src/components/dashboard/main-content/course-enrollments/course-cards/mark-complete-modal/MarkCompleteModal.jsx

```diff
@@ -73,26 +75,40 @@ const MarkCompleteModal = ({
 
   return (
     <MarkCompleteModalContext.Provider value={contextValue}>
-      <Modal
+      <ModalDialog
         title="Save course for later"
-        body={<ModalBody />}
-        buttons={[
-          <StatefulButton
-            labels={{
-              default: MARK_SAVED_FOR_LATER_DEFAULT_LABEL,
-              pending: MARK_SAVED_FOR_LATER_PENDING_LABEL,
-            }}
-            disabledStates={['pending']}
-            className="confirm-mark-complete-btn btn-brand-primary"
-            state={confirmButtonState}
-            onClick={handleConfirmButtonClick}
-            key="confirm-mark-complete-btn"
-          />,
-        ]}
-        open={isOpen && !confirmSuccessful}
+        isOpen={isOpen && !confirmSuccessful}
         onClose={handleModalOnClose}
-        closeText="Cancel"
-      />
+        hasCloseButton
+        isFullscreenOnMobile
+      >
+        <ModalDialog.Header>
+          <ModalDialog.Title>
+            Save course for later
+          </ModalDialog.Title>
+        </ModalDialog.Header>
+        <ModalDialog.Body>
+          <ModalBody />
+        </ModalDialog.Body>
+        <ModalDialog.Footer>
+          <ActionRow>
+            <ModalDialog.CloseButton variant="tertiary" data-testid="mark-complete-modal-cancel-btn">
+              Cancel
+            </ModalDialog.CloseButton>
+            <StatefulButton
+              labels={{
+                default: MARK_SAVED_FOR_LATER_DEFAULT_LABEL,
+                pending: MARK_SAVED_FOR_LATER_PENDING_LABEL,
+              }}
+              disabledStates={['pending']}
+              className="confirm-mark-complete-btn btn-brand-primary"
+              state={confirmButtonState}
+              onClick={handleConfirmButtonClick}
+              key="confirm-mark-complete-btn"
```

Comment:
nit: the `key` isnt needed since the component isnt being rendered within a list (e.g., within a `.map`). Likely a copy/paste from Paragon docs site.